src/comp/RSchedule.hs

```
@@ -267,11 +282,11 @@ colorFw rMax dropEdges st g
 
 
 -- backward pass: pick up vertices and color them
-colorBk :: [Integer] -> [(UniqueUse,Integer)] -> [RRM] -> StkL ->
-           ErrorMonad ([(UniqueUse, Integer)], [RRM])
+colorBk :: [Integer] -> M.Map UniqueUse Integer -> [RRM] -> StkL ->
+           ErrorMonad (M.Map UniqueUse Integer, [RRM])
 colorBk _  cs es [] = return (cs,es)
 colorBk rMaxL cs es (Vertex vns@(v,_) : vs) =
-    colorBk rMaxL ((v, pickColor rMaxL cs vns):cs) es vs
+    colorBk rMaxL (M.insert v (pickColor rMaxL cs vns) cs) es vs
```

This silently replaces an existing key.  In keeping with your use of internal errors elsewhere, did you want to use an insert function reports an internal error when the key exists?

---

I don't feel strongly.  I can go with whatever you want.

A comment wouldn't hurt.  I assume there's no performance impact to using a function with an error.

---

I ended up taking out the internalError for rSchedule and switching to a comment because I thought the code looked nicer and closer to what was there before.